### PR TITLE
Feature/sync/527 update archie dependencies

### PR DESCRIPTION
**Pull request by jakesmolka**
## Changes

- Archie changes
    - Migrated to Archie > 1.0.0 and its strict invariant checks (**Important**: might break working systems, due to possibility to now rejecting data which worked before.)
- Test data changes
    - Changed Party_Ref type: `ANY` to `PARTY`
    - Added archetype_details to composition's content root
    - Changed terminology ID `"value": "Unicode"` to `"value": "IANA_character-sets"`
    - Changed DV_PROPORTION, mostly precision to 1 if one value had one decimal place
    - Added or changed null_flavour on ELEMENTs
    - Changed DV_EHR_URI values to match `ehr://...` schema

## Related issue

EHRbase side of:
https://github.com/ehrbase/project_management/issues/527

## Additional information and checks

- [x] Pull request linked in changelog

### CODE REVIEW

File: tests/robot/_resources/test_data_sets/contributions/invalid/minimal/minimal_admin.contribution.modification.json [hunk 1]
```diff
@@ -121,6 +121,18 @@
             "_type": "DV_TEXT",
             "value": "Minimal"
           },
+          "archetype_details": {
```
Comment (ppazos) [inline tests/robot/_resources/test_data_sets/contributions/invalid/minimal/minimal_admin.contribution.modification.json:124]: @jakesmolka archetype_details should be optional https://specifications.openehr.org/releases/RM/Release-1.1.0/common.html#_overview_2

File: tests/robot/_resources/test_data_sets/contributions/invalid/minimal_persistent/minimal_persistent.contribution.modification.json [hunk 2]
```diff
@@ -102,6 +102,18 @@
       			"_type": "DV_TEXT",
       			"value": "Minimal"
       		},
+					"archetype_details": {
```
Comment (ppazos) [inline tests/robot/_resources/test_data_sets/contributions/invalid/minimal_persistent/minimal_persistent.contribution.modification.json:105]: @jakesmolka archetype_details should be optional https://specifications.openehr.org/releases/RM/Release-1.1.0/common.html#_overview_2

File: tests/robot/_resources/test_data_sets/contributions/invalid/ref_to_non_existent_OPT.json [hunk 3]
```diff
@@ -373,6 +385,18 @@
       			"_type": "DV_TEXT",
       			"value": "Minimal"
       		},
+					"archetype_details": {
```
Comment (ppazos) [inline tests/robot/_resources/test_data_sets/contributions/invalid/ref_to_non_existent_OPT.json:388]: @jakesmolka archetype_details should be optional https://specifications.openehr.org/releases/RM/Release-1.1.0/common.html#_overview_2

File: tests/robot/_resources/test_data_sets/invalid_templates/minimal_persistent/persistent_minimal.composition.extdatetime.invalid.xml [hunk 4]
```diff
@@ -44,6 +44,15 @@
     </composer>
     <content archetype_node_id="openEHR-EHR-OBSERVATION.minimal.v1" xsi:type="OBSERVATION">
       <!-- removed mandatory name -->
+      <archetype_details>
```
Comment (ppazos) [inline tests/robot/_resources/test_data_sets/invalid_templates/minimal_persistent/persistent_minimal.composition.extdatetime.invalid.xml:47]: @jakesmolka archetype_details should be optional https://specifications.openehr.org/releases/RM/Release-1.1.0/common.html#_overview_2

**Review by ppazos — CHANGES_REQUESTED** [review]

**Review by ppazos — APPROVED** [review]
:shipit:

File: README.md [hunk 5]
```diff
@@ -4,6 +4,11 @@ EHRbase is an [openEHR](openehr.org) Clinical Data Repository, providing a stand
 
 ## Release Notes
 
+##### WIP: 2021, XYZ 00
+v?.?.? - (...)
+
+**Important:** Please note that this release introduces [Archie's](https://github.com/openEHR/archie) new strict invariant checks. Depending on existing data and clients this might be a breaking change. Please carefully check the EHRbase output and update your input data if EHRbase rejects it. The strict validation can also be deactivated via configuration, but caution in advised!
+
```
[the remark this answers is not in the document]
Comment (jakesmolka) [reply README.md:11]: I'll add that part to the documentation.

**Pull request merged**